webui/react/src/pages/TrialDetails/Profiles/SystemMetricChart.tsx

```diff
@@ -29,10 +29,10 @@ const SystemMetricChart: React.FC<Props> = ({ filters, trial }: Props) => {
     filters.gpuUuid,
   );
 
-  useEffect(() => {
-    if (!chartRef.current) return;
+  const xMin = chartData && chartData[0] && chartData[0][0] ? chartData[0][0] : 0;
```

Comment:
**suggestion:** `useMemo` might be enough to avoid an actual state declaration and a need for useEffect:
```
const { chartData, xMin } = useMemo(() => {
  const chartData = convertMetricsToUplotData(systemMetrics.dataByUnixTime);
  const xMin = chartData && chartData[0] && chartData[0][0] ? chartData[0][0] : 0;
  return { chartData, xMin };
}, [ systemMetrics.dataByUnixTime ]);
```